der/derive/src/attributes.rs

```diff
@@ -101,6 +109,12 @@ impl FieldAttrs {
                 }
 
                 tag_mode = Some(mode);
+            } else if let Some(default_attr_str) = attr.parse_value::<String>("default") {
+                if default.is_some() {
+                    abort!(attr.value, "duplicate ASN.1 `tag_mode` attribute");
+                }
+
+                default = Some(Ident::new(&default_attr_str, attrs[counter].span()));
```

Comment:
This works for the `Ident` case, although I think to do this "properly" it should be a `Path`.

I think you can use the [`Parse`](https://docs.rs/syn/latest/syn/parse/trait.Parse.html) trait to parse a `Path` from the `LitStr` (retaining span info automatically), and perhaps the `value` portion of `AttrNameValue` should retain the original `LitStr`.